CONTRIBUTING.rst

```diff
@@ -464,9 +487,97 @@ Step 5: Pass PR Review
 Note that committers will use **Squash and Merge** instead of **Rebase and Merge**
 when merging PRs and your commit will be squashed to single commit.
 
-You need to have review of at least one committer (if you are committer yourself, it has to be
+When the maintainer starts conversations it is expected that you respond to questions, suggestions, doubts,
+and generally it's great if all such conversations seem to converge to a common understanding. You do not
+necessarily have to apply all the suggestions (often they are just opinions and suggestions even if they are
+coming from seasoned maintainers) - it's perfectly ok that you respond to it with your own opinions and
+understanding of the problem and your approach and if you have good arguments, presenting them is a good idea.
+
+The reviewers might leave several types of responses:
+
+* ``General PR comment`` - which usually means that there is a question/opinion/suggestion on how the PR can be
+  improved, or it's an ask to explain how you understand the PR. You can usually quote some parts of such
+  general comment and respond to it in your comments. Often comments that are raising questions in general
+  might lead to different discussions, even a request to move the discussion to the devlist or even lead to
+  completely new PRs created as a spin-off of the discussion.
+
+* ``Comment/Conversation around specific lines of code`` - such conversation usually flags a potential
+  improvement, or a potential problem with the code. It's a good idea to respond to such comments and explain
+  your approach and understanding of the problem. If you do not agree with the comment, you can explain why
+  you think your approach is better. If you agree with the comment, you can explain why you did not do it
```

Comment:
```suggestion
  you think your approach is better. If you agree with the comment, you can optionally explain why you did not do it
```
Can we make this optional, please? IMO, I would generally not have thought of the approach in the first place or I would have been naive/unaware/beginner.

Reply:
I think I tried to be too detailed here. I rephrased this whole chapter a bit.